state/cockroachdb/cockroachdb_access.go

```diff
@@ -386,7 +401,25 @@ func (p *cockroachDBAccess) Query(req *state.QueryRequest) (*state.QueryResponse
 
 // Ping implements database ping.
 func (p *cockroachDBAccess) Ping() error {
-	return p.db.Ping()
+	retryCount := defaultMaxConnectionAttempts
+	if p.metadata.MaxConnectionAttempts != nil && *p.metadata.MaxConnectionAttempts >= 0 {
+		retryCount = *p.metadata.MaxConnectionAttempts
+	}
+	publishBo := backoff.NewExponentialBackOff()
```

Comment:
Please do not import `backoff/v4` directly. Instead use this from `dapr/kit/retry`. There is a `PolicyExponential` convenience option, but can also specify the specifics.